src/preferences/colorpaletteeditormodel.cpp

```diff
@@ -117,24 +132,78 @@ void ColorPaletteEditorModel::setColorPalette(const ColorPalette& palette) {
     setDirty(false);
 }
 
-ColorPalette ColorPaletteEditorModel::getColorPalette(const QString& name) const {
+ColorPalette ColorPaletteEditorModel::getColorPalette(
+        const QString& name) const {
     QList<mixxx::RgbColor> colors;
     QMap<int, int> hotcueColorIndices;
     for (int i = 0; i < rowCount(); i++) {
         QStandardItem* pColorItem = item(i, 0);
-        QStandardItem* pHotcueIndexItem = item(i, 1);
-        mixxx::RgbColor::optional_t color = mixxx::RgbColor::fromQString(pColorItem->text());
+        HotcueIndexListItem* pHotcueIndexItem =
+                static_cast<HotcueIndexListItem>(item(i, 1));
+        mixxx::RgbColor::optional_t color =
+                mixxx::RgbColor::fromQString(pColorItem->text());
         if (color) {
+            QList<int> hotcueIndexes =
+                    parseHotcueIndexList(pHotcueIndexItem->text());
             colors << *color;
 
-            bool ok;
-            int hotcueIndex = pHotcueIndexItem->text().toInt(&ok);
-            if (ok) {
-                hotcueColorIndices.insert(hotcueIndex - 1, colors.size() - 1);
+            for (int i = 0; i < hotcueIndexes.size(); ++i) {
+                hotcueColorIndices.insert(
+                        hotcueIndexes.at(i) - 1, colors.size() - 1);
             }
         }
     }
     // If we have a non consequitive list of hotcue indexes, indexes are shifted down
     // due to the sorting nature of QMap. This is intended, this way we have a color for every hotcue.
     return ColorPalette(name, colors, hotcueColorIndices.values());
 }
+
+HotcueIndexListItem::HotcueIndexListItem(const QList<int>& hotcueList = {})
+        : QStandardItem(), m_hotcueIndexList(hotcueList) {
+}
+QVariant HotcueIndexListItem::data(int role = Qt::UserRole + 1) const {
+    QStringList serializedIndexStrings;
+    for (const int index : m_hotcueIndexList) {
+        serializedIndexStrings.append(QString::number(index));
+    }
+    return QVariant(serializedIndexStrings);
+}
+
+void HotcueIndexListItem::setData(
+        const QVariant& value, int role = Qt::UserRole + 1) {
+    m_hotcueIndexList.clear();
+    // QRegularExpressionMatchIterator regexResults = HotcueIndexListItem::hotcueListMatchingRegex.globalMatch(value.toString());
+    // while (regexResults.hasNext())
+    // {
+    //     QRegularExpressionMatch match = regexResults.next();
+    //     bool ok;
+    //     const int hotcueIndex = match.captured(1).toInt(&ok);
+    //     if (ok) {
+    //         m_hotcueIndexList.append(hotcueIndex);
+    //     }
+    // }
+    const auto hotcueIndexStrings = value.toStringList();
```

Comment:
I'm just using a stringlist here because I wanted to test whether it was possible to somehow use that. Though I simply can't know since I don't fully understand how QVariant is supposed to work with a stringlist in this context. The Qt docs don't mention that.